src/main/java/com/disi/TravelPoints/service/WishlistService.java

```diff
@@ -27,6 +31,27 @@ public List<LandmarkDetails> getWishlistForUser(Long id) throws Exception {
                 .collect(Collectors.toList());
     }
 
+    public Long addLandmarkInWishlist(Long landmarkId, Long userId) throws Exception {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new Exception("User not found!"));
+        Landmark landmark = landmarkRepository.findById(landmarkId)
+                .orElseThrow(() -> new Exception("Landmark not found!"));
+
+        Optional<Wishlist> wishlist = wishlistRepository.findByUser_Id(userId);
+        if (wishlist.isEmpty()) {
+            Wishlist newWishlist = new Wishlist();
+            newWishlist.setUser(user);
+            newWishlist.setLandmarks(Set.of(landmark));
+            return wishlistRepository.save(newWishlist).getId();
+        }
+
+        Wishlist actualWishlist = wishlist.get();
+        final Set<Landmark> landmarks = actualWishlist.getLandmarks();
+        landmarks.add(landmark);
+        actualWishlist.setLandmarks(landmarks);
```

Comment:
Shouldn't you also update the database after updating the local context?

Reply:
You are right, I was in a hurry